### PR TITLE
[IMPROVEMENT] Use autoFocus rather than timeout + focus

### LINKED ISSUE #1067: [IMPROVEMENT] Use autoFocus rather than timeout + focus
In some places, text input must be focused automatically on did mount, but we could do it better.
Currently, we're using a setTimeout to wait for navigation transition animation and then a `inputRef.focus()`.
`TextInput` has a `autoFocus` prop that does it.

Example: [`LoginView`](https://github.com/RocketChat/Rocket.Chat.ReactNative/blob/develop/app/views/LoginView.js#L88)
```jsx
// We could remove this entire block and add `autoFocus` on username input and remove unnecessary ref
componentDidMount() {
	this.timeout = setTimeout(() => {
		this.usernameInput.focus();
	}, 600);
}
```

### Where to do it:
- [ ] CreateChannelView
- [ ] ForgotPasswordView
- [ ] LoginView
- [ ] NewServerView
- [ ] RegisterView
- [ ] SearchMessagesView
- [ ] SetUsernameView

### PR DESCRIPTION
**Pull request by tj4752**
@RocketChat/ReactNative

Closes #1067

### CODE REVIEW

**Comment by diegolmello** [comment]
Thanks for your PR, @tj4752!
It looks ok, but I'm going to review it properly only next week.
We're finishing last steps of 1.17.0 and we're going to release it by monday/tuesday.
Cheers.

File: app/views/SetUsernameView.js [hunk 1]
```diff
@@ -114,6 +111,7 @@ export default class SetUsernameView extends React.Component {
 						<Text style={[sharedStyles.loginSubtitle, sharedStyles.textRegular]}>{I18n.t('Set_username_subtitle')}</Text>
 						<TextInput
 							inputRef={e => this.usernameInput = e}
```
Comment (IlarionHalushka) [inline app/views/SetUsernameView.js]: remove inputRef

File: app/views/SearchMessagesView/index.js [hunk 2]
```diff
@@ -137,6 +133,7 @@ export default class SearchMessagesView extends React.Component {
 				<View style={styles.searchContainer}>
 					<RCTextInput
 						inputRef={(e) => { this.name = e; }}
```
Comment (IlarionHalushka) [inline app/views/SearchMessagesView/index.js]: remove inputRef

File: app/views/RegisterView.js [hunk 3]
```diff
@@ -207,6 +201,7 @@ export default class RegisterView extends React.Component {
 						<Text style={[sharedStyles.loginTitle, sharedStyles.textBold]}>{I18n.t('Sign_Up')}</Text>
 						<TextInput
 							inputRef={(e) => { this.nameInput = e; }}
```
Comment (IlarionHalushka) [inline app/views/RegisterView.js]: remove inputRef

File: app/views/NewServerView.js [hunk 4]
```diff
@@ -169,6 +165,7 @@ export default class NewServerView extends React.Component {
 						<Text style={styles.title}>{I18n.t('Sign_in_your_server')}</Text>
 						<TextInput
 							inputRef={e => this.input = e}
```
Comment (IlarionHalushka) [inline app/views/NewServerView.js]: remove inputRef

File: app/views/SearchMessagesView/index.js (hunk 2 above)
Comment (IlarionHalushka) [inline app/views/SearchMessagesView/index.js]: remove inputRef

File: app/views/ForgotPasswordView.js [hunk 5]
```diff
@@ -100,6 +94,7 @@ export default class ForgotPasswordView extends React.Component {
 						<Text style={[sharedStyles.loginTitle, sharedStyles.textBold]}>{I18n.t('Forgot_password')}</Text>
 						<TextInput
 							inputRef={(e) => { this.emailInput = e; }}
```
Comment (IlarionHalushka) [inline app/views/ForgotPasswordView.js]: remove inputRef

File: app/views/CreateChannelView.js [hunk 6]
```diff
@@ -338,6 +335,7 @@ export default class CreateChannelView extends React.Component {
 						<View style={sharedStyles.separatorVertical}>
 							<TextInput
 								ref={ref => this.channelNameRef = ref}
```
Comment (IlarionHalushka) [inline app/views/CreateChannelView.js]: remove inputRef

**Comment by diegolmello** [comment]
@tj4752 Merged. Thank you! 👊

**Pull request merged**